### ISSUE TITLE
[Bug]: One pod stuck, cannot reconcile

### ISSUE THREAD

**Issue by Ornias1993** [cloudnative-pg/cloudnative-pg#3974]
### Is there an existing issue already for this bug?

- [X] I have searched for an existing issue, and could not find anything. I believe this is a new bug.

### I have read the troubleshooting guide

- [X] I have read the troubleshooting guide and I think this is a new bug.

### I am running a supported version of CloudNativePG

- [X] I have read the troubleshooting guide and I think this is a new bug.

### Contact Details

info@kjeldschouten.nl

### Version

1.22.1

### What version of Kubernetes are you using?

1.28

### What is your Kubernetes environment?

Other

### How did you install the operator?

Helm

### What happened?

Environment: 
TALOS Linux

Pod-1 does not start, it throws a reconciler error.
Deleting either this one, or both, results in the new pod still getting stuck.

Pod-2 runs fine, however.

### Cluster resource

```shell
apiVersion: postgresql.cnpg.io/v1
kind: Cluster
metadata:
  annotations:
    cnpg.io/hibernation: 'off'
    meta.helm.sh/release-name: bitmagnet
    meta.helm.sh/release-namespace: media
    rollme: dT0F0
  creationTimestamp: '2024-02-28T22:52:33Z'
  generation: 2
  labels:
    app: bitmagnet-1.3.0
    app.kubernetes.io/instance: bitmagnet
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: bitmagnet
    app.kubernetes.io/version: 0.7.7
    cnpg.io/reload: 'on'
    helm-revision: '2'
    helm.sh/chart: bitmagnet-1.3.0
    helm.toolkit.fluxcd.io/name: bitmagnet
    helm.toolkit.fluxcd.io/namespace: media
    release: bitmagnet
  managedFields:
    - apiVersion: postgresql.cnpg.io/v1
      fieldsType: FieldsV1
      fieldsV1:
        f:metadata:
          f:annotations:
            .: {}
            f:cnpg.io/hibernation: {}
            f:meta.helm.sh/release-name: {}
            f:meta.helm.sh/release-namespace: {}
            f:rollme: {}
          f:labels:
            .: {}
            f:app: {}
            f:app.kubernetes.io/instance: {}
            f:app.kubernetes.io/managed-by: {}
            f:app.kubernetes.io/name: {}
            f:app.kubernetes.io/version: {}
            f:cnpg.io/reload: {}
            f:helm-revision: {}
            f:helm.sh/chart: {}
            f:helm.toolkit.fluxcd.io/name: {}
            f:helm.toolkit.fluxcd.io/namespace: {}
            f:release: {}
        f:spec:
          .: {}
          f:bootstrap:
            .: {}
            f:initdb:
              .: {}
              f:dataChecksums: {}
              f:database: {}
              f:owner: {}
              f:secret:
                .: {}
                f:name: {}
          f:enableSuperuserAccess: {}
          f:failoverDelay: {}
          f:instances: {}
          f:logLevel: {}
          f:maxSyncReplicas: {}
          f:minSyncReplicas: {}
          f:monitoring:
            f:disableDefaultQueries: {}
            f:enablePodMonitor: {}
          f:nodeMaintenanceWindow:
            .: {}
            f:inProgress: {}
            f:reusePVC: {}
          f:postgresGID: {}
          f:postgresUID: {}
          f:primaryUpdateMethod: {}
          f:primaryUpdateStrategy: {}
          f:replicationSlots:
            .: {}
            f:highAvailability:
              .: {}
              f:enabled: {}
              f:slotPrefix: {}
            f:updateInterval: {}
          f:resources:
            .: {}
            f:limits:
              .: {}
              f:cpu: {}
              f:memory: {}
            f:requests:
              .: {}
              f:cpu: {}
              f:memory: {}
          f:smartShutdownTimeout: {}
          f:startDelay: {}
          f:stopDelay: {}
          f:storage:
            .: {}
            f:pvcTemplate:
              .: {}
              f:accessModes: {}
              f:resources:
                .: {}
                f:requests:
                  .: {}
                  f:storage: {}
              f:storageClassName: {}
            f:resizeInUseVolumes: {}
          f:switchoverDelay: {}
          f:walStorage:
            .: {}
            f:pvcTemplate:
              .: {}
              f:accessModes: {}
              f:resources:
                .: {}
                f:requests:
                  .: {}
                  f:storage: {}
              f:storageClassName: {}
            f:resizeInUseVolumes: {}
      manager: helm-controller
      operation: Update
      time: '2024-03-01T11:08:25Z'
    - apiVersion: postgresql.cnpg.io/v1
      fieldsType: FieldsV1
      fieldsV1:
        f:status:
          .: {}
          f:certificates:
            .: {}
            f:clientCASecret: {}
            f:expirations:
              .: {}
              f:bitmagnet-cnpg-main-ca: {}
              f:bitmagnet-cnpg-main-replication: {}
              f:bitmagnet-cnpg-main-server: {}
            f:replicationTLSSecret: {}
            f:serverAltDNSNames: {}
            f:serverCASecret: {}
            f:serverTLSSecret: {}
          f:cloudNativePGCommitHash: {}
          f:cloudNativePGOperatorHash: {}
          f:conditions: {}
          f:configMapResourceVersion:
            .: {}
            f:metrics:
              .: {}
              f:cnpg-default-monitoring: {}
          f:currentPrimary: {}
          f:currentPrimaryTimestamp: {}
          f:healthyPVC: {}
          f:instanceNames: {}
          f:instances: {}
          f:instancesReportedState:
            .: {}
            f:bitmagnet-cnpg-main-1:
              .: {}
              f:isPrimary: {}
            f:bitmagnet-cnpg-main-2:
              .: {}
              f:isPrimary: {}
              f:timeLineID: {}
          f:instancesStatus:
            .: {}
            f:healthy: {}
            f:replicating: {}
          f:latestGeneratedNode: {}
          f:managedRolesStatus: {}
          f:phase: {}
          f:phaseReason: {}
          f:poolerIntegrations:
            .: {}
            f:pgBouncerIntegration:
              .: {}
              f:secrets: {}
          f:pvcCount: {}
          f:readService: {}
          f:readyInstances: {}
          f:secretsResourceVersion:
            .: {}
            f:applicationSecretVersion: {}
            f:clientCaSecretVersion: {}
            f:replicationSecretVersion: {}
            f:serverCaSecretVersion: {}
            f:serverSecretVersion: {}
            f:superuserSecretVersion: {}
          f:targetPrimary: {}
          f:targetPrimaryTimestamp: {}
          f:timelineID: {}
          f:topology:
            .: {}
            f:instances:
              .: {}
              f:bitmagnet-cnpg-main-1: {}
              f:bitmagnet-cnpg-main-2: {}
            f:nodesUsed: {}
            f:successfullyExtracted: {}
          f:writeService: {}
      manager: manager
      operation: Update
      subresource: status
      time: '2024-03-01T14:26:24Z'
  name: bitmagnet-cnpg-main
  namespace: media
  resourceVersion: '53402236'
  uid: 13795c92-1c23-4f07-8e60-0bfc3ef0f544
  selfLink: /apis/postgresql.cnpg.io/v1/namespaces/media/clusters/bitmagnet-cnpg-main
status:
  certificates:
    clientCASecret: bitmagnet-cnpg-main-ca
    expirations:
      bitmagnet-cnpg-main-ca: 2024-05-28 22:47:33 +0000 UTC
      bitmagnet-cnpg-main-replication: 2024-05-28 22:47:33 +0000 UTC
      bitmagnet-cnpg-main-server: 2024-05-28 22:47:33 +0000 UTC
    replicationTLSSecret: bitmagnet-cnpg-main-replication
    serverAltDNSNames:
      - bitmagnet-cnpg-main-rw
      - bitmagnet-cnpg-main-rw.media
      - bitmagnet-cnpg-main-rw.media.svc
      - bitmagnet-cnpg-main-r
      - bitmagnet-cnpg-main-r.media
      - bitmagnet-cnpg-main-r.media.svc
      - bitmagnet-cnpg-main-ro
      - bitmagnet-cnpg-main-ro.media
      - bitmagnet-cnpg-main-ro.media.svc
    serverCASecret: bitmagnet-cnpg-main-ca
    serverTLSSecret: bitmagnet-cnpg-main-server
  cloudNativePGCommitHash: c7be872e
  cloudNativePGOperatorHash: 262d86af058f59462fdccaec51231a1afd888153413f63224f60e458fcd335be
  conditions:
    - lastTransitionTime: '2024-03-01T10:16:52Z'
      message: Cluster Is Not Ready
      reason: ClusterIsNotReady
      status: 'False'
      type: Ready
    - lastTransitionTime: '2024-02-28T22:52:49Z'
      message: Continuous archiving is working
      reason: ContinuousArchivingSuccess
      status: 'True'
      type: ContinuousArchiving
  configMapResourceVersion:
    metrics:
      cnpg-default-monitoring: '52121470'
  currentPrimary: bitmagnet-cnpg-main-2
  currentPrimaryTimestamp: '2024-03-01T11:58:24.876337Z'
  healthyPVC:
    - bitmagnet-cnpg-main-1
    - bitmagnet-cnpg-main-1-wal
    - bitmagnet-cnpg-main-2
    - bitmagnet-cnpg-main-2-wal
  instanceNames:
    - bitmagnet-cnpg-main-1
    - bitmagnet-cnpg-main-2
  instances: 2
  instancesReportedState:
    bitmagnet-cnpg-main-1:
      isPrimary: false
    bitmagnet-cnpg-main-2:
      isPrimary: true
      timeLineID: 2
  instancesStatus:
    healthy:
      - bitmagnet-cnpg-main-2
    replicating:
      - bitmagnet-cnpg-main-1
  latestGeneratedNode: 2
  managedRolesStatus: {}
  phase: Waiting for the instances to become active
  phaseReason: Some instances are not yet active. Please wait.
  poolerIntegrations:
    pgBouncerIntegration:
      secrets:
        - bitmagnet-cnpg-main-pooler
  pvcCount: 4
  readService: bitmagnet-cnpg-main-r
  readyInstances: 1
  secretsResourceVersion:
    applicationSecretVersion: '53286816'
    clientCaSecretVersion: '52217929'
    replicationSecretVersion: '52217942'
    serverCaSecretVersion: '52217929'
    serverSecretVersion: '52217938'
    superuserSecretVersion: '52217943'
  targetPrimary: bitmagnet-cnpg-main-2
  targetPrimaryTimestamp: '2024-03-01T11:58:21.041881Z'
  timelineID: 2
  topology:
    instances:
      bitmagnet-cnpg-main-1: {}
      bitmagnet-cnpg-main-2: {}
    nodesUsed: 2
    successfullyExtracted: true
  writeService: bitmagnet-cnpg-main-rw
spec:
  affinity:
    podAntiAffinityType: preferred
  bootstrap:
    initdb:
      dataChecksums: false
      database: bitmagnet
      encoding: UTF8
      localeCType: C
      localeCollate: C
      owner: bitmagnet
      secret:
        name: bitmagnet-cnpg-main-user
  enableSuperuserAccess: true
  failoverDelay: 0
  imageName: ghcr.io/cloudnative-pg/postgresql:16.1
  instances: 2
  logLevel: info
  maxSyncReplicas: 0
  minSyncReplicas: 0
  monitoring:
    customQueriesConfigMap:
      - key: queries
        name: cnpg-default-monitoring
    disableDefaultQueries: true
    enablePodMonitor: true
  nodeMaintenanceWindow:
    inProgress: false
    reusePVC: true
  postgresGID: 26
  postgresUID: 26
  postgresql:
    parameters:
      archive_mode: 'on'
      archive_timeout: 5min
      dynamic_shared_memory_type: posix
      log_destination: csvlog
      log_directory: /controller/log
      log_filename: postgres
      log_rotation_age: '0'
      log_rotation_size: '0'
      log_truncate_on_rotation: 'false'
      logging_collector: 'on'
      max_parallel_workers: '32'
      max_replication_slots: '32'
      max_worker_processes: '32'
      shared_memory_type: mmap
      shared_preload_libraries: ''
      ssl_max_protocol_version: TLSv1.3
      ssl_min_protocol_version: TLSv1.3
      wal_keep_size: 512MB
      wal_receiver_timeout: 5s
      wal_sender_timeout: 5s
    syncReplicaElectionConstraint:
      enabled: false
  primaryUpdateMethod: switchover
  primaryUpdateStrategy: unsupervised
  replicationSlots:
    highAvailability:
      enabled: true
      slotPrefix: _cnpg_
    updateInterval: 30
  resources:
    limits:
      cpu: '4'
      memory: 8Gi
    requests:
      cpu: 10m
      memory: 50Mi
  smartShutdownTimeout: 180
  startDelay: 3600
  stopDelay: 1800
  storage:
    pvcTemplate:
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          storage: 100Gi
      storageClassName: openebs-hostpath
    resizeInUseVolumes: true
  switchoverDelay: 3600
  walStorage:
    pvcTemplate:
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          storage: 100Gi
      storageClassName: openebs-hostpath
    resizeInUseVolumes: true
```

### Relevant log output

```shell
{"level":"info","ts":"2024-03-01T14:24:51Z","logger":"postgres","msg":"record","logging_pod":"bitmagnet-cnpg-main-1","record":{"log_time":"2024-03-01 14:24:51.943 UTC","process_id":"687","session_id":"65e1e533.2af","session_line_num":"8","session_start_time":"2024-03-01 14:24:51 UTC","transaction_id":"0","error_severity":"LOG","sql_state_code":"00000","message":"aborting any active transactions","backend_type":"postmaster","query_id":"0"}}
{"level":"info","ts":"2024-03-01T14:24:51Z","logger":"postgres","msg":"record","logging_pod":"bitmagnet-cnpg-main-1","record":{"log_time":"2024-03-01 14:24:51.945 UTC","process_id":"687","session_id":"65e1e533.2af","session_line_num":"9","session_start_time":"2024-03-01 14:24:51 UTC","transaction_id":"0","error_severity":"LOG","sql_state_code":"00000","message":"background worker \"logical replication launcher\" (PID 695) exited with exit code 1","backend_type":"postmaster","query_id":"0"}}
{"level":"info","ts":"2024-03-01T14:24:51Z","logger":"postgres","msg":"record","logging_pod":"bitmagnet-cnpg-main-1","record":{"log_time":"2024-03-01 14:24:51.946 UTC","process_id":"689","session_id":"65e1e533.2b1","session_line_num":"1","session_start_time":"2024-03-01 14:24:51 UTC","transaction_id":"0","error_severity":"LOG","sql_state_code":"00000","message":"shutting down","backend_type":"checkpointer","query_id":"0"}}
{"level":"info","ts":"2024-03-01T14:24:51Z","logger":"postgres","msg":"record","logging_pod":"bitmagnet-cnpg-main-1","record":{"log_time":"2024-03-01 14:24:51.964 UTC","process_id":"689","session_id":"65e1e533.2b1","session_line_num":"2","session_start_time":"2024-03-01 14:24:51 UTC","transaction_id":"0","error_severity":"LOG","sql_state_code":"00000","message":"checkpoint starting: shutdown immediate","backend_type":"checkpointer","query_id":"0"}}
{"level":"info","ts":"2024-03-01T14:24:52Z","logger":"wal-archive","msg":"Backup not configured, skip WAL archiving","logging_pod":"bitmagnet-cnpg-main-1","walName":"pg_wal/000000010000001700000051","currentPrimary":"bitmagnet-cnpg-main-2","targetPrimary":"bitmagnet-cnpg-main-2"}
{"level":"info","ts":"2024-03-01T14:24:52Z","logger":"postgres","msg":"record","logging_pod":"bitmagnet-cnpg-main-1","record":{"log_time":"2024-03-01 14:24:52.009 UTC","process_id":"689","session_id":"65e1e533.2b1","session_line_num":"3","session_start_time":"2024-03-01 14:24:51 UTC","transaction_id":"0","error_severity":"LOG","sql_state_code":"00000","message":"checkpoint complete: wrote 2 buffers (0.0%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.001 s, sync=0.002 s, total=0.048 s; sync files=2, longest=0.002 s, average=0.001 s; distance=16384 kB, estimate=16384 kB; lsn=17/52000028, redo lsn=17/52000028","backend_type":"checkpointer","query_id":"0"}}
{"level":"info","ts":"2024-03-01T14:24:52Z","logger":"postgres","msg":"record","logging_pod":"bitmagnet-cnpg-main-1","record":{"log_time":"2024-03-01 14:24:52.015 UTC","process_id":"687","session_id":"65e1e533.2af","session_line_num":"10","session_start_time":"2024-03-01 14:24:51 UTC","transaction_id":"0","error_severity":"LOG","sql_state_code":"00000","message":"database system is shut down","backend_type":"postmaster","query_id":"0"}}
{"level":"info","ts":"2024-03-01T14:24:52Z","logger":"pg_ctl","msg":"waiting for server to shut down.... done","pipe":"stdout","logging_pod":"bitmagnet-cnpg-main-1"}
{"level":"info","ts":"2024-03-01T14:24:52Z","logger":"pg_ctl","msg":"server stopped","pipe":"stdout","logging_pod":"bitmagnet-cnpg-main-1"}
{"level":"info","ts":"2024-03-01T14:24:52Z","msg":"Exited log pipe","fileName":"/controller/log/postgres.csv","logging_pod":"bitmagnet-cnpg-main-1"}
{"level":"info","ts":"2024-03-01T14:24:52Z","msg":"Extracting pg_controldata information","controller":"cluster","controllerGroup":"postgresql.cnpg.io","controllerKind":"Cluster","Cluster":{"name":"bitmagnet-cnpg-main","namespace":"media"},"namespace":"media","name":"bitmagnet-cnpg-main","reconcileID":"0345b73b-9105-4f06-afa8-c0336ab88e38","uuid":"77459c42-d7d7-11ee-bcdc-e6d90df483b8","logging_pod":"bitmagnet-cnpg-main-1","reason":"before pg_rewind"}
{"level":"info","ts":"2024-03-01T14:24:52Z","logger":"pg_controldata","msg":"pg_control version number:            1300\nCatalog version number:               202307071\nDatabase system identifier:           7340789579528605717\nDatabase cluster state:               shut down\npg_control last modified:             Fri 01 Mar 2024 02:24:52 PM UTC\nLatest checkpoint location:           17/52000028\nLatest checkpoint's REDO location:    17/52000028\nLatest checkpoint's REDO WAL file:    000000010000001700000052\nLatest checkpoint's TimeLineID:       1\nLatest checkpoint's PrevTimeLineID:   1\nLatest checkpoint's full_page_writes: on\nLatest checkpoint's NextXID:          0:106552\nLatest checkpoint's NextOID:          90114\nLatest checkpoint's NextMultiXactId:  820\nLatest checkpoint's NextMultiOffset:  1639\nLatest checkpoint's oldestXID:        722\nLatest checkpoint's oldestXID's DB:   1\nLatest checkpoint's oldestActiveXID:  0\nLatest checkpoint's oldestMultiXid:   1\nLatest checkpoint's oldestMulti's DB: 1\nLatest checkpoint's oldestCommitTsXid:0\nLatest checkpoint's newestCommitTsXid:0\nTime of latest checkpoint:            Fri 01 Mar 2024 02:24:51 PM UTC\nFake LSN counter for unlogged rels:   0/3E8\nMinimum recovery ending location:     0/0\nMin recovery ending loc's timeline:   0\nBackup start location:                0/0\nBackup end location:                  0/0\nEnd-of-backup record required:        no\nwal_level setting:                    logical\nwal_log_hints setting:                on\nmax_connections setting:              100\nmax_worker_processes setting:         32\nmax_wal_senders setting:              10\nmax_prepared_xacts setting:           0\nmax_locks_per_xact setting:           64\ntrack_commit_timestamp setting:       off\nMaximum data alignment:               8\nDatabase block size:                  8192\nBlocks per segment of large relation: 131072\nWAL block size:                       8192\nBytes per WAL segment:                16777216\nMaximum length of identifiers:        64\nMaximum columns in an index:          32\nMaximum size of a TOAST chunk:        1996\nSize of a large-object chunk:         2048\nDate/time type storage:               64-bit integers\nFloat8 argument passing:              by value\nData page checksum version:           0\nMock authentication nonce:            10329c8174df5ad528760c6ad8ac8d7771349615a751dd345ba7051d7bcf4a75\n","pipe":"stdout","logging_pod":"bitmagnet-cnpg-main-1"}
{"level":"info","ts":"2024-03-01T14:24:52Z","msg":"Starting up pg_rewind","controller":"cluster","controllerGroup":"postgresql.cnpg.io","controllerKind":"Cluster","Cluster":{"name":"bitmagnet-cnpg-main","namespace":"media"},"namespace":"media","name":"bitmagnet-cnpg-main","reconcileID":"0345b73b-9105-4f06-afa8-c0336ab88e38","uuid":"77459c42-d7d7-11ee-bcdc-e6d90df483b8","logging_pod":"bitmagnet-cnpg-main-1","pgdata":"/var/lib/postgresql/data/pgdata","options":["-P","--source-server","host=bitmagnet-cnpg-main-rw user=streaming_replica port=5432 sslkey=/controller/certificates/streaming_replica.key sslcert=/controller/certificates/streaming_replica.crt sslrootcert=/controller/certificates/server-ca.crt application_name=bitmagnet-cnpg-main-1 sslmode=verify-ca dbname=postgres","--target-pgdata","/var/lib/postgresql/data/pgdata","--restore-target-wal"]}
{"level":"info","ts":"2024-03-01T14:24:52Z","logger":"pg_rewind","msg":"pg_rewind: connected to server","pipe":"stderr","logging_pod":"bitmagnet-cnpg-main-1"}
{"level":"info","ts":"2024-03-01T14:24:52Z","logger":"pg_rewind","msg":"pg_rewind: servers diverged at WAL location 14/78913D98 on timeline 1","pipe":"stderr","logging_pod":"bitmagnet-cnpg-main-1"}
{"level":"info","ts":"2024-03-01T14:24:52Z","logger":"wal-restore","msg":"tried restoring WALs, but no backup was configured","logging_pod":"bitmagnet-cnpg-main-1"}
{"level":"info","ts":"2024-03-01T14:24:52Z","logger":"pg_rewind","msg":"pg_rewind: error: could not restore file \"000000010000001400000077\" from archive","pipe":"stderr","logging_pod":"bitmagnet-cnpg-main-1"}
{"level":"info","ts":"2024-03-01T14:24:52Z","logger":"pg_rewind","msg":"pg_rewind: error: could not find previous WAL record at 14/77FFF8C0","pipe":"stderr","logging_pod":"bitmagnet-cnpg-main-1"}
{"level":"error","ts":"2024-03-01T14:24:52Z","msg":"Failed to execute pg_rewind","controller":"cluster","controllerGroup":"postgresql.cnpg.io","controllerKind":"Cluster","Cluster":{"name":"bitmagnet-cnpg-main","namespace":"media"},"namespace":"media","name":"bitmagnet-cnpg-main","reconcileID":"0345b73b-9105-4f06-afa8-c0336ab88e38","uuid":"77459c42-d7d7-11ee-bcdc-e6d90df483b8","logging_pod":"bitmagnet-cnpg-main-1","options":["-P","--source-server","host=bitmagnet-cnpg-main-rw user=streaming_replica port=5432 sslkey=/controller/certificates/streaming_replica.key sslcert=/controller/certificates/streaming_replica.crt sslrootcert=/controller/certificates/server-ca.crt application_name=bitmagnet-cnpg-main-1 sslmode=verify-ca dbname=postgres","--target-pgdata","/var/lib/postgresql/data/pgdata","--restore-target-wal"],"error":"exit status 1","stacktrace":"github.com/cloudnative-pg/cloudnative-pg/pkg/management/log.(*logger).Error\n\tpkg/management/log/log.go:128\ngithub.com/cloudnative-pg/cloudnative-pg/pkg/management/postgres.(*Instance).Rewind\n\tpkg/management/postgres/instance.go:977\ngithub.com/cloudnative-pg/cloudnative-pg/internal/management/controller.(*InstanceReconciler).verifyPgDataCoherenceForPrimary\n\tinternal/management/controller/instance_startup.go:259\ngithub.com/cloudnative-pg/cloudnative-pg/internal/management/controller.(*InstanceReconciler).initialize\n\tinternal/management/controller/instance_controller.go:349\ngithub.com/cloudnative-pg/cloudnative-pg/internal/management/controller.(*InstanceReconciler).Reconcile\n\tinternal/management/controller/instance_controller.go:138\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:119\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:316\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:227"}
{"level":"error","ts":"2024-03-01T14:24:52Z","msg":"Reconciler error","controller":"cluster","controllerGroup":"postgresql.cnpg.io","controllerKind":"Cluster","Cluster":{"name":"bitmagnet-cnpg-main","namespace":"media"},"namespace":"media","name":"bitmagnet-cnpg-main","reconcileID":"0345b73b-9105-4f06-afa8-c0336ab88e38","error":"error executing pg_rewind: exit status 1","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:329\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:227"}
{"level":"info","ts":"2024-03-01T14:24:54Z","msg":"Unable to collect metrics","logging_pod":"bitmagnet-cnpg-main-1","error":"failed to connect to `host=/controller/run user=postgres database=postgres`: dial error (dial unix /controller/run/.s.PGSQL.5432: connect: no such file or directory)"}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by gbartolini** [comment]
This happens because you don't have a WAL archive and you are not using synchronous replication (you only have a standby). You need to manually destroy the PVC.

**Comment by Ornias1993** [comment]
> This happens because you don't have a WAL archive and you are not using synchronous replication (you only have a standby). You need to manually destroy the PVC.

How can that happen? this was healthy before..

**Comment by Ornias1993** [comment]
for correct self-healing I think synchronous replication is required, not?

**Comment by mnencia** [comment]
While Gabriele is correct in suggesting you set up a backup in your cluster, this specific issue could have been caused by a bug fixed by #3930 and #4132.

To be sure about the root cause, I would need the labels of Pods and PVCs and the complete status of the Cluster resource, but unfortunately, they are missing here, so I'm closing this issue.

**Comment by Ornias1993** [comment]
> To be sure about the root cause, I would need the labels of Pods and PVCs and the complete status of the Cluster resource, but unfortunately, they are missing here, so I'm closing this issue.

Its more than a tad impolite to close issues for things that are missing without asking those first.

And you got the complete cluster-resource, including status... There isn't any more to give other than a complete dump of the cluster resource imho.
